### ISSUE TITLE
‘GCM_IV_DATA_LEN’ undeclared

### ISSUE THREAD

**Issue by Edelweiss014** [intel/QAT_Engine#259]
Hi there,

I am trying to build QAT engine with ipsec_mb. I found the source code of ipsec_mb here. 
I used the option `--with-qat_sw_ipsec_mb_install_dir=/usr/include` (because I found intel-ipsec-mb.h here after installing ipsec_mb). 
But I still got this error:
```
make  all-am
make[1]: Entering directory '/root/QAT_Engine'
  CC       qat_evp.lo
qat_evp.c: In function ‘qat_create_gcm_cipher_meth’:
qat_evp.c:758:49: error: ‘GCM_IV_DATA_LEN’ undeclared (first use in this function); did you mean ‘IMB_GCM_IV_DATA_LEN’?
  758 |         res &= EVP_CIPHER_meth_set_iv_length(c, GCM_IV_DATA_LEN);
      |                                                 ^~~~~~~~~~~~~~~
      |                                                 IMB_GCM_IV_DATA_LEN
qat_evp.c:758:49: note: each undeclared identifier is reported only once for each function it appears in
make[1]: *** [Makefile:1259: qat_evp.lo] Error 1
make[1]: Leaving directory '/root/QAT_Engine'
make: *** [Makefile:917: all] Error 2
```

I checked intel-ipsec-mb.h and found only IMB_GCM_IV_DATA_LEN there. But I do not know where GCM_IV_DATA_LEN should be. 

Can anyone help me with this? Thanks a lot!

**Comment by amitk18x** [comment]
Support for IMB_GCM_IV_DATA_LEN is added and will be released in the next release.  Meanwhile, you may use tags/v1.3 of intel-ipsec to build qatengine.

**Comment by ColinIanKing** [comment]
this fixes the issue for me:

```
Index: QAT_Engine-1.2.0/qat_evp.c
===================================================================
--- QAT_Engine-1.2.0.orig/qat_evp.c
+++ QAT_Engine-1.2.0/qat_evp.c
@@ -755,7 +755,7 @@ const EVP_CIPHER *qat_create_gcm_cipher_

 #ifdef ENABLE_QAT_SW_GCM
     if (qat_sw_offload && (qat_sw_algo_enable_mask & ALGO_ENABLE_MASK_AES_GCM)) {
-        res &= EVP_CIPHER_meth_set_iv_length(c, GCM_IV_DATA_LEN);
+        res &= EVP_CIPHER_meth_set_iv_length(c, IMB_GCM_IV_DATA_LEN);
         res &= EVP_CIPHER_meth_set_flags(c, VAESGCM_FLAG);
 #ifndef QAT_OPENSSL_PROVIDER
         res &= EVP_CIPHER_meth_set_init(c, vaesgcm_ciphers_init);
```

**Comment by mythi** [comment]
> this fixes the issue for me:

I can confirm it works: https://github.com/intel/intel-device-plugins-for-kubernetes/pull/1478/commits/0c490ea45282a56060911c8e8529d74b96fb8e07

**Comment by Yogaraj-Alamenda** [comment]
Fixed in v1.3.0 release
